src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java

```diff
@@ -357,18 +358,11 @@ public Resource getResource(ResolveContext<JcrProviderState> ctx, String path, R
                     version = child.getResourceMetadata().getParameterMap().get("v");
                 }
                 if (version == null) {
-                    Item item = ((JcrItemResource<?>) child).getItem();
-                    if ("/".equals(item.getPath())) {
+                    Node parentNode = ((JcrItemResource)child).getParentNode();
+                    if (parentNode == null) {
                         return null;
                     }
-                    Node parentNode;
-                    try {
-                        parentNode = item.getParent();
-                    } catch(AccessDeniedException e) {
-                        return null;
-                    }
-                    String parentPath = ResourceUtil.getParent(child.getPath());
-                    return new JcrNodeResource(ctx.getResourceResolver(), parentPath, version, parentNode,
+                    return new JcrNodeResource(ctx.getResourceResolver(), parentNode.getPath(), null, parentNode,
```

Comment:
The current version is not using Node.getPath to avoid that a RepositoryException can be thrown (in theory).
So maybe it makes sense to refactor your PR and as a first thing, get the parentPath, then call a method that tries to get the node for that path (similar to your getParentNode method but just getting a path as an argument) and then reuse parentPath here

Reply:
ho missed that one! so i guess i can leave JcrItemResource alone, thanks!